src/test/java/de/tum/in/www1/artemis/CourseIntegrationTest.java

```diff
@@ -71,6 +98,95 @@ public void getCourseWithoutPermission() throws Exception {
         request.getList("/api/courses", HttpStatus.FORBIDDEN, Course.class);
     }
 
+    @Test
+    @WithMockUser(username = "student1", roles = "USER")
+    public void getAllCoursesForDashboard() throws Exception {
+        prepareGetAllCoursesForDashboard();
+        List<Course> courses = request.getList("/api/courses/for-dashboard", HttpStatus.OK, Course.class);
+
+        assertThat(courses.size()).as("Inactive course was filtered out").isEqualTo(1);
+        assertThat(courses.get(0).getExercises().size()).as("Two exercises are returned").isEqualTo(2);
+
+        for (Exercise exercise : courses.get(0).getExercises()) {
```

Comment:
This is complicated stuff, a lot of nested loops, ifs, etc.
- Needs comments to explain what the checks mean
- Could be split up into single methods

Reply:
I've only added comments because I don't know how to split it up properly, as I'm using each previous assertion (or at least the conditionals) for null checks.